cpp/src/neighbors/ivf_flat/ivf_flat_extend_float_int64_t.cu

```diff
@@ -0,0 +1,71 @@
+    ->cuvs::neighbors::ivf_flat::index<T, IdxT>                                                  \
+  {                                                                                              \
+    return cuvs::neighbors::ivf_flat::index<T, IdxT>(std::move(                                  \
+      cuvs::neighbors::ivf_flat::detail::extend(handle, new_vectors, new_indices, orig_index))); \
+  }                                                                                              \
+                                                                                                 \
+  void extend(raft::resources const& handle,                                                     \
+              raft::device_matrix_view<const T, IdxT, raft::row_major> new_vectors,              \
+              std::optional<raft::device_vector_view<const IdxT, IdxT>> new_indices,             \
+              cuvs::neighbors::ivf_flat::index<T, IdxT>* idx)                                    \
+  {                                                                                              \
+    cuvs::neighbors::ivf_flat::detail::extend(handle, new_vectors, new_indices, idx);            \
+  }                                                                                              \
+  auto extend(raft::resources const& handle,                                                     \
+              raft::host_matrix_view<const T, IdxT, raft::row_major> new_vectors,                \
+              std::optional<raft::host_vector_view<const IdxT, IdxT>> new_indices,               \
+              const cuvs::neighbors::ivf_flat::index<T, IdxT>& orig_index)                       \
+    ->cuvs::neighbors::ivf_flat::index<T, IdxT>                                                  \
+  {                                                                                              \
+    return cuvs::neighbors::ivf_flat::index<T, IdxT>(std::move(                                  \
+      cuvs::neighbors::ivf_flat::detail::extend(handle, new_vectors, new_indices, orig_index))); \
+  }                                                                                              \
+                                                                                                 \
+  void extend(raft::resources const& handle,                                                     \
+              raft::host_matrix_view<const T, IdxT, raft::row_major> new_vectors,                \
+              std::optional<raft::host_vector_view<const IdxT, IdxT>> new_indices,               \
+              cuvs::neighbors::ivf_flat::index<T, IdxT>* idx)                                    \
+  {                                                                                              \
+    cuvs::neighbors::ivf_flat::detail::extend(handle, new_vectors, new_indices, idx);            \
+  }
+CUVS_INST_IVF_FLAT_EXTEND(float, int64_t);
```

Comment:
Since `ivf_flat::build()` is calling `ivf_flat::extend()`, those two functions (with the same type) can be in the same compilation unit. I saw a compilation time and size improvement in IVF-PQ when doing this change.